backend/src/main/kotlin/be/osoc/team1/backend/services/EmailService.kt

```diff
@@ -0,0 +1,89 @@
+package be.osoc.team1.backend.services
+
+import be.osoc.team1.backend.exceptions.InvalidGmailCredentialsException
+import org.springframework.mail.MailAuthenticationException
+import org.springframework.mail.SimpleMailMessage
+import org.springframework.mail.javamail.JavaMailSender
+import org.springframework.mail.javamail.JavaMailSenderImpl
+import org.springframework.stereotype.Service
+import java.util.Properties
+import java.util.UUID
+
+/**
+ * This class contains every function needed to make and send emails.
+ */
+@Service
+class EmailService {
+    /**
+     * Credentials of gmail account to send emails with.
+     */
+    private var emailAddressSender: String? = System.getenv("OSOC_GMAIL_ADDRESS")
+    private var passwordSender: String? = System.getenv("OSOC_GMAIL_APP_PASSWORD")
+
+    /**
+     * Set credentials of account to send mails with. This function is used for testing.
+     */
+    fun setSenderEmailCredentials(email: String?, password: String?) {
+        emailAddressSender = email
+        passwordSender = password
+    }
+
+    /**
+     * Make the body of the email users receive when they request a password change.
+     */
+    private fun getForgotPasswordEmailBody(forgotPasswordUUID: UUID): String {
+        val url = "https://sel2-1.ugent.be/forgotPassword/$forgotPasswordUUID"
+        return """
+            Hi,
+            
+            Trouble signing in? Resetting your password is easy.
+            Use the link below to choose a new password.
+            You can only use this link once to reset your password and it is only valid for 20 minutes.
+            $url
+            (if this link isn't clickable, you can copy and paste it into the search bar)
+            
+            If you did not forget your password, please disregard this email.
+        """.trimIndent()
+    }
+
+    /**
+     * Get a [JavaMailSender] object which is correctly configured.
+     */
+    private fun getMailSender(): JavaMailSender {
```

Comment:
Alright, I hope I'm not too annoying with all the changes I'm requesting but I think this could be moved into the constructor by using ```init { }```, mailSender would just become a field, so sendEmail would just do `mailSender.send()` instead of always creating a new `MailSender` instance (unless there is a valid reason to always create a new mail sender?). As you have probably noticed by now, I'm just trying to make the code as object-oriented as possible, but we are nearly there.

Reply:
Well, believe it or not but I just changed that before reading your comment, so I definitely agree. I also agree with your other requests and I think it's valuable that you give feedback.
So please don't hesitate or feel guilty when pointing out what I could improve on, feedback like that improves our code and I might as well learn something from it.